File: app/src/main/java/com/jdjv/smsbot/BotRules.java
package com.jdjv.smsbot;

import java.util.Calendar;

public class BotRules {

    private static final String[][] RULES = {
        {"سلام|درود|hello|hi",       "سلام! من SmsBat هستم. چطور می‌تونم کمکت کنم؟"},
        {"ساعت|وقت|time",            null},   // dynamic
        {"اسمت|کی هستی|who",         "من SmsBat هستم، یه بات خودکار اس‌ام‌اس."},
        {"کمک|help|راهنما",          "می‌تونی بپرسی: سلام، ساعت، اسمت چیه، ممنون"},
        {"ممنون|مرسی|thanks|خوبی",   "خواهش می‌کنم! کار دیگه‌ای هست؟"},
        {"خداحافظ|بای|bye",          "خداحافظ! موفق باشی."},
    };

    public static String getReply(String message) {
        String lower = message.trim().toLowerCase();
        for (String[] rule : RULES) {
            String[] keywords = rule[0].split("\\|");
            for (String kw : keywords) {
                if (lower.contains(kw)) {
                    if (rule[1] == null) {
                        Calendar cal = Calendar.getInstance();
                        return String.format("الان ساعت %02d:%02d هست.",
                            cal.get(Calendar.HOUR_OF_DAY), cal.get(Calendar.MINUTE));
                    }
                    return rule[1];
                }
            }
        }
        return "پیامت رو گرفتم: \"" + message + "\"\nبرای راهنما بنویس: کمک";
    }
}

File: app/src/main/java/com/jdjv/smsbot/BotRules.kt
package com.jdjv.smsbot

object BotRules {

    data class Rule(val keywords: List<String>, val reply: String)

    private val rules = listOf(
        Rule(listOf("سلام", "درود", "hello", "hi"),
            "سلام! من یه بات هستم. چطور می‌تونم کمکت کنم؟"),

        Rule(listOf("ساعت", "وقت", "چند", "time"),
            "الان ساعت ${currentTime()} هست."),

        Rule(listOf("اسمت", "اسم", "کی هستی", "who"),
            "من SmsBat هستم، یه بات خودکار اس‌ام‌اس."),

        Rule(listOf("کمک", "help", "راهنما"),
            "می‌تونی بپرسی: سلام، ساعت، وقت آزاد، اسمت چیه"),

        Rule(listOf("ممنون", "مرسی", "thanks", "خوبی"),
            "خواهش می‌کنم! کار دیگه‌ای هست؟"),

        Rule(listOf("خداحافظ", "بای", "bye"),
            "خداحافظ! موفق باشی.")
    )

    fun getReply(message: String): String {
        val lower = message.trim().lowercase()
        for (rule in rules) {
            if (rule.keywords.any { lower.contains(it) }) {
                return rule.reply
            }
        }
        return "پیامت رو گرفتم: \"$message\"\nمتاسفانه جواب این رو نمی‌دونم. برای کمک بنویس: کمک"
    }

    private fun currentTime(): String {
        val cal = java.util.Calendar.getInstance()
        return String.format("%02d:%02d", cal.get(java.util.Calendar.HOUR_OF_DAY), cal.get(java.util.Calendar.MINUTE))
    }
}
